x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/mapper/DenseVectorFieldMapper.java

```diff
@@ -170,44 +209,70 @@ public boolean parsesArrayValue() {
 
     @Override
     public void parse(DocumentParserContext context) throws IOException {
-        int dims = fieldType().dims(); //number of vector dimensions
+        if (context.doc().getByKey(fieldType().name()) != null) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() +
+                "] doesn't not support indexing multiple values for the same field in the same document");
+        }
 
+        Field field = fieldType().indexed
+            ? parseKnnVector(context)
+            : parseBinaryDocValuesVector(context);
+        context.doc().addWithKey(fieldType().name(), field);
+    }
+
+    private Field parseKnnVector(DocumentParserContext context) throws IOException {
+        float[] vector = new float[dims];
```

Comment:
it would be nice to  create float[] vector = new float[dims] once in DenseVectorFieldMapper constructor and they reuse it instead of creating a new array each time during parsing. Especially considering that Lucene's `VectorValuesWriter` would anyway make a separate copy:  vectors.add(ArrayUtil.copyOfSubArray(vectorValue, 0, vectorValue.length)).  But I don't know how if all this is done within a single thread.

Reply:
I guess we can keep an eye on this in indexing benchmarks. Like you said, the indexing logic often copies field values, so I don't expect this array creation to be a bottleneck.